app/models/notification.rb

```diff
@@ -0,0 +1,20 @@
+class Notification < ApplicationRecord
+  belongs_to :notification_type
+  belongs_to :initiator, :class_name => User, :foreign_key => 'user_id'
+  belongs_to :subject, :polymorphic => true
+  belongs_to :cause, :polymorphic => true
+  has_many :notification_recipients, :dependent => :delete_all
+  has_many :recipients, :class_name => User, :through => :notification_recipients
+
+  def self.emit(type, opts = {})
+    type = NotificationType.find_by_name!(type)
+    notification = create(opts.merge!(:notification_type => type))
+    [notification, notification.create_recipients]
+  end
+
+  def create_recipients
+    subscribers = notification_type.subscriber_ids(subject, initiator)
+    notification_recipients.create! subscribers.collect { |id| {:user_id => id } }
```

Comment:
Great idea. :fireworks: When we use nested attributes the whole operation will be put into a single sql transaction. When as opposed to running `create []` which runs n+1 sql transactions. :cupid: The code will be a little more cryptic, but what can I do, I need to listen to the reviewer. :person_frowning: